core/server/api/shared/http.js

```diff
@@ -14,9 +14,8 @@ const http = (apiImpl) => {
             params: req.params,
             user: req.user,
             context: {
-                user: ((req.user && req.user.id) || (req.user && models.User.isExternalUser(req.user.id))) ? req.user.id : null,
-                client: (req.client && req.client.slug) ? req.client.slug : null,
-                client_id: (req.client && req.client.id) ? req.client.id : null
+                api_key: (req.api_key && req.api_key.id) ? req.api_key.id : null,
```

Comment:
I'd suggest either we assign the whole object or we assign `api_key_id`?

Reply:
Really? That's not what we do for the user, but can rename if you want

Reply:
Yeah it isn't 👍 , but the naming for `context.user` was in the first place not the right choice (probably happened years ago), because 

1. You sometimes need access to the whole user
2. You expect `context.user` to be an object and not an id.

Reply:
Yeah - this is true. Okay instead of renaming, I think I should just attach the whole key - what do you think?

Reply:
Actually - on second thoughts the context object is primarily for the permissions service, at that expects ids for most things - so `api_key_id` is probably best

Reply:
If we use `api_key_id`, we could still attach `api_key_somethingelse` 👍